### PR TITLE
Textbox: add floating label support (#503)

### LINKED ISSUE #503: Textbox: add floating label support
Let's leverage Skin's `label` module to add floating label support to the textbox component.

For example:

<ebay-textbox floating-label="Email address" />

Related ticket: https://github.com/eBay/skin/issues/497

### PR DESCRIPTION
**Pull request by mkchang**
## Description

Add support for floating label for ebay-textbox. 

## Context

Allow ebay-textbox to leverage existing Skin styling. 

Make use of [makeup-floating-label](https://github.com/makeup-js/makeup-floating-label) in ebay-textbox to toggle necessary classes and create floating effect. This is the same package already used on the Skin GitHub Page to demo Label/floating label.

## References

Closes #503 
Jira Ticket: https://jirap.corp.ebay.com/browse/EBAYUI-189
Supporting skin update for DS4 support: https://github.com/eBay/skin/pull/508

### DS4
image

### DS6
image

### CODE REVIEW

**Comment by seangates** [comment]
For posterity:
image

I don't know how to fix it, though. I think Google's implementation even has the same problem.

File: src/components/ebay-textbox/index.js [hunk 1]
```diff
@@ -28,14 +37,22 @@ function getInitialState(input) {
         iconPostfix,
         tag: input.fluid ? 'div' : 'span',
         textboxTag: Boolean(input.multiline) ? 'textarea' : 'input',
-        invalid: String(Boolean(input.invalid))
+        invalid: String(Boolean(input.invalid)),
```
Comment (seangates) [inline src/components/ebay-textbox/index.js]: Some of these values are evaluated here in the return, while others are above. Should we consolidate them all to above now, to make things just a little more readable and easier to follow?

**Comment by ianmcburnie** [comment]
Sean reminds me that we expose a `refresh` method in the floating-label module, as a means to refresh the ui in the case of browser autofill kicking in like this.

File: src/components/ebay-textbox/index.js (hunk 1 above)
Comment (mkchang) [reply src/components/ebay-textbox/index.js]: Are you thinking something like this?
```
    function getInitialState(input) {
        const { floatingLabel, icon, style } = input;
        const iconTag = input.iconTag && input.iconTag.renderBody;
        const invalid = String(Boolean(input.invalid));
        const tag = input.fluid ? 'div' : 'span';
        const textboxTag = Boolean(input.multiline) ? 'textarea' : 'input';
        const classes = ['textbox__control'];
        const displayIcon = Boolean(input.icon && !input.multiline && input.iconTag);
        const iconPostfix = input.iconPosition === 'postfix';
        const iconPrefix = input.iconPosition === 'prefix' || !iconPostfix;
        const rootClass = ['textbox', input.class];
        const labelClasses = ['floating-label__label'];
        const htmlAttributes = processHtmlAttributes(input);
        if (input.fluid) {
            classes.push('textbox__control--fluid');
        }
        if (input.floatingLabel) {
            classes.push('textbox__control--underline');
        }
        if (displayIcon && iconPostfix) {
            rootClass.push('textbox--icon-end');
        }
        if (htmlAttributes.disabled) {
            labelClasses.push('floating-label__label--disabled');
        }

        return {
            htmlAttributes,
            rootClass,
            style,
            classes,
            icon,
            iconTag,
            displayIcon,
            iconPrefix,
            iconPostfix,
            tag,
            textboxTag,
            invalid,
            floatingLabel,
            labelClasses
        };
    }
```
Wasn't sure if changing lines around so much (consolidating consts and if statements) constitutes a broken window or not, but let me know what you think. Could destructure even more of the `input.x` make it look cleaner by dropping the `input.` part, though maybe at cost of ease to understand now that the link to input is removed.

**Comment by mkchang** [comment]
Sounds good, thanks! @ianmcburnie let me know when 2.0.0 is pointing to the Skin v7 prerelease and I can update this branch then before the squash.

**Pull request merged**